website/source/guides/load-balancing/haproxy.html.md

```diff
@@ -0,0 +1,272 @@
+application.
+
+HAProxy version 1.8+ (LTS) includes server-template[server-template] which lets
+users specify placeholder backend servers to populate HAProxy’s load balancing
+pools. Server-template can use Consul as one of these backend servers,
+requesting SRV records from Consul DNS.
+
+## Reference Material
+
+- [HAProxy][haproxy]
+- [Load Balancing Strategies for Consul][lb-strategies]
+
+## Estimated Time to Complete
+
+20 minutes
+
+## Prerequisites
+
+To perform the tasks described in this guide, you need to have a Nomad
+environment with Consul installed. You can use this [repo][terraform-repo] to
+easily provision a sandbox environment. This guide will assume a cluster with
+one server node and three client nodes.
+
+-> **Note:** This guide is for demo purposes and is only using a single
+server node. In a production cluster, 3 or 5 server nodes are recommended.
+
+## Steps
+
+### Step 1: Create a Job for Demo Web App
+
+Create a job for a demo web application and name the file 'webapp.nomad'
+
+```hcl
+job "demo-webapp" {
+  datacenters = ["dc1"]
+
+  group "demo" {
+    count = 3
+
+    task "server" {
+      env {
+        PORT    = "${NOMAD_PORT_http}"
+        NODE_IP = "${NOMAD_IP_http}"
+      }
+
+      driver = "docker"
+
+      config {
+        image = "hashicorp/demo-webapp-lb-guide"
+      }
+
+      resources {
+        network {
+          mbits = 10
+          port  "http"{}
+        }
+      }
+
+      service {
+        name = "demo-webapp"
+        port = "http"
+
+        check {
+          type     = "http"
+          path     = "/"
+          interval = "2s"
+          timeout  = "2s"
+        }
+      }
+    }
+  }
+}
+```
+
+Note that this job deploys 3 instances of our demo web application which we will
+load balance with HAProxy in the next few steps.
+
+### Step 2: Deploy the Demo Web App
+
+We can now deploy our demo web application:
+
+```shell
+$ nomad run webapp.nomad 
+==> Monitoring evaluation "8f3af425"
+    Evaluation triggered by job "demo-webapp"
+    Evaluation within deployment: "dc4c1925"
+    Allocation "bf9f850f" created: node "d16a11fb", group "demo"
+    Allocation "25e0496a" created: node "b78e27be", group "demo"
+    Allocation "a97e7d39" created: node "01d3eb32", group "demo"
+    Evaluation status changed: "pending" -> "complete"
+==> Evaluation "8f3af425" finished with status "complete"
+```
+
+### Step 3: Create a Job for HAProxy
+
+Create a job for HAProxy and name it `haproxy.nomad`. This will be our load
+balancer that will balance requests to the deployed instances of our web
+application.
+
+```hcl
+job "haproxy" {
+  region      = "global"
+  datacenters = ["dc1"]
+  type        = "service"
+
+  group "haproxy" {
+    count = 1
+
+    task "haproxy" {
+      driver = "docker"
+
+      config {
+        image        = "haproxy:2.0"
+        network_mode = "host"
+
+        port_map {
+          http = 8080
+        }
```

Comment:
it it worthwhile/necessary to add 1936 to this and the resources stanza below?

Reply:
Just added this for clarity. Thanks.